### PR TITLE
add image element/s

### PR DESCRIPTION
**Pull request by KazuCocoa**
experimental yet

based on https://github.com/appium/python-client/pull/224/files

- [x] run tests by manual
- [x] implement tests
- [x] docstring
- [x] update readme

### CODE REVIEW

File: lib/appium_lib_core/common/base/driver.rb [hunk 1]
```diff
@@ -183,6 +183,45 @@ def get_timeouts
         def session_capabilities
           @bridge.session_capabilities
         end
+
+        DEFAULT_MATCH_THRESHOLD = 0.5
+        def find_element_by_image(png_img_path, match_threshol = DEFAULT_MATCH_THRESHOLD)
```
Comment (mykola-mokhnach) [inline lib/appium_lib_core/common/base/driver.rb]: match_threshol -> match_threshold

File: lib/appium_lib_core/common/base/driver.rb [hunk 2]
```diff
@@ -183,6 +183,45 @@ def get_timeouts
...
+          full_image = self.screenshot_as(:base64)
+
+          options = {}
+          options[:goodMatchesFactor] = (match_threshol * 100).to_i unless match_threshol.nil?
+
+          params = {}
+          params[:mode] = :matchTemplate
+          params[:firstImage] = full_image
+          params[:secondImage] = File.read(png_img_path)
```
Comment (mykola-mokhnach) [inline lib/appium_lib_core/common/base/driver.rb]: shouldn't it be base64-encodeed first?

File: lib/appium_lib_core/element/image.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+module Appium
+  module Core
+    class ImageElement
+      Point     = Struct.new(:x, :y)
+      Dimension = Struct.new(:width, :height)
+      Rectangle = Struct.new(:x, :y, :width, :height)
+      Location  = Struct.new(:latitude, :longitude, :altitude)
```
Comment (mykola-mokhnach) [inline lib/appium_lib_core/element/image.rb]: I assume it's not THAT location ;)

File: lib/appium_lib_core/element/image.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+      @data = {}
+
+      def initialize(driver, x, y, width, height)
+        @driver = driver
+
+        @data[:center_x] = (x + width) / 2
```
Comment (mykola-mokhnach) [inline lib/appium_lib_core/element/image.rb]: are you sure that parentheses are properly placed?

File: lib/appium_lib_core/element/image.rb (hunk 3 above)
Comment (KazuCocoa) [reply lib/appium_lib_core/element/image.rb]: Yes, this is a different thing.
I'm comparing WebDriver's Element class to handle as same as possible and this comes from it.
(I'll clean up after some testing and refactoring.)

File: lib/appium_lib_core/element/image.rb (hunk 4 above)
Comment (jlipps) [reply lib/appium_lib_core/element/image.rb]: should be `x + (width / 2)` (same below)

**Comment by KazuCocoa** [comment]
With https://github.com/appium/ruby_lib_core/pull/95/files?short_path=c3bce62#diff-c3bce62ecaa34e736cf9eefc272dba78
image

**Pull request merged**